app/src/main/java/ch/epfl/cs311/wanderwave/model/spotify/SpotifyController.kt

```diff
@@ -315,3 +319,100 @@ constructor(
     FAILED
   }
 }
+/**
+ * Get all the element of the main screen and add them to the top list
+ *
+ * @author Menzo Bouaissi
+ * @since 2.0
+ * @last update 3.0
+ */
+fun retrieveAndAddSubsectionFromSpotify(
+    spotifySubsectionList: MutableStateFlow<List<ListItem>>,
+    spotifyController: SpotifyController,
+    scope: CoroutineScope
+) {
+  scope.launch {
+    val track = spotifyController.getAllElementFromSpotify().firstOrNull()
+    if (track != null) {
+      for (i in track) {
+        spotifySubsectionList.value += i
+      }
+    }
+  }
+}
+
+/**
+ * Get all the element of the main screen and add them to the top list
+ *
+ * @author Menzo Bouaissi
+ * @since 2.0
+ * @last update 3.0
+ */
+fun retrieveChildFromSpotify(
```

Comment:
This function and the one above look extremely similar, do you think you could find a way to "merge" them to avoid duplication ? It's really not a big deal but it'd be nice to have

Reply:
Actually, they are not the same function. As you can see, one is calling  getAllElementFromSpotify and the other one getAllChildren(item). I can try to modularize the fact of adding element into the list. If you see a better way of merge it you can show me